dkg/frostp2p.go

```diff
@@ -32,7 +32,11 @@ var (
 )
 
 // newFrostP2P returns a p2p frost transport implementation.
-func newFrostP2P(tcpNode host.Host, peers map[peer.ID]cluster.NodeIdx, secret *k1.PrivateKey, threshold int) *frostP2P {
+func newFrostP2P(tcpNode host.Host, peers map[peer.ID]cluster.NodeIdx, secret *k1.PrivateKey, threshold int) (*frostP2P, error) {
+	if secret == nil {
+		return nil, errors.New("secret cannot be nil")
+	}
```

Comment:
I'm not worried about this, as the code should ensure not to pass in nil.

Reply:
cool, I'll revert this one